api/org/commcare/api/persistence/SqlIndexedStorageUtility.java

```diff
@@ -0,0 +1,388 @@
+            System.out.println("SqlIndexedStorageUtility readBytes exception: " + e);
+        } finally {
+            try {
+                if (c != null) {
+                    c.close();
+                }
+                if (preparedStatement != null) {
+                    preparedStatement.close();
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+        return false;
+    }
+
+    @Override
+    public Object getAccessLock() {
+        // TODO Auto-generated method stub
+        return null;
+    }
+
+    @Override
+    public int getNumRecords() {
+        PreparedStatement preparedStatement = null;
+        Connection c = null;
+        try {
+            c = getConnection();
+            String sqlQuery = "SELECT COUNT (*) FROM " + this.tableName + ";";
+            preparedStatement = c.prepareStatement(sqlQuery);
+            ResultSet rs = preparedStatement.executeQuery();
+            return rs.getInt(1);
+        } catch (Exception e) {
+            System.out.println("SqlIndexedStorageUtility readBytes exception: " + e);
+        } finally {
+            try {
+                if (c != null) {
+                    c.close();
+                }
+                if (preparedStatement != null) {
+                    preparedStatement.close();
+                }
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+        return -1;
+    }
+
+    @Override
+    public int getRecordSize(int id) {
+        return 0;
+    }
+
+    @Override
+    public JdbcSqlStorageIterator<T> iterate() {
+        Connection connection;
+        ResultSet resultSet = null;
+        PreparedStatement preparedStatement = null;
+        try {
+            connection = this.getConnection();
+            String sqlQuery = "SELECT " + org.commcare.modern.database.DatabaseHelper.ID_COL + " , " +
+                    org.commcare.modern.database.DatabaseHelper.DATA_COL + " FROM " + this.tableName + ";";
+            preparedStatement = connection.prepareStatement(sqlQuery);
+            resultSet = preparedStatement.executeQuery();
+        } catch (SQLException | ClassNotFoundException e) {
+            e.printStackTrace();
+        }
+        return new JdbcSqlStorageIterator<T>(preparedStatement, resultSet, this.getNumRecords(), this);
+    }
+
+    @Override
+    public int getTotalSize() {
+        //serialize and test blah blah.
+        return 0;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return this.getNumRecords() <= 0;
+    }
+
+    @Override
+    public byte[] readBytes(int id) {
+        PreparedStatement preparedStatement = null;
+        Connection c = null;
+        try {
+            c = getConnection();
+            preparedStatement = SqlHelper.prepareIdSelectStatement(c, this.tableName, id);
+            if (preparedStatement == null) {
+                return null;
```

Comment:
the prepared statement being invalid does not mean that the bytes should be null